### ISSUE TITLE
Updating Multiple Properties at Once

### ISSUE THREAD

**Issue by mouse0270** [cferdinandi/reef#175]
I have an object that looks like this:
```javascript
let CONTROL = store({
	value: 4,
	options: [{value: 4, text: 'New' }, {value: 1, text: 'Closed'}]
}, 'Status')
```
I then have an event listener that looks like this:
```javascript
document.addEventListener('reef:store-Status', async (event) => {
	console.log('Status Changed', event.detail);
});
```
After looking around at the docs and issues here, I found this issue https://github.com/cferdinandi/reef/issues/47 in which you stated that you batch updates so that it would only be called once....

But when doing this
```javascript
CONTROLS.Status.options = [
	{ value: 1, text: 'Open' }, 
	{ value: 2, text: 'Closed' }, 
	{ value: 3, text: 'Forwarded' }, 
	{ value: 4, text: 'Cancelled' }
];
CONTROLS.Status.value = ElementHistory.ElementStatus;
```

I get two console log statements... How should I be batching these changes so they happen together instead of seperately?

**Comment by mouse0270** [comment]
Here is an modified example of one of your examples to show this issue, if you look at the console, you'll see its called twice and not batched together.

https://codepen.io/mouse0270/pen/LYJdJLB?editors=1111

**Comment by Sh1eld** [comment]
@mouse0270 (responding to the edit)

https://reefjs.com/api/#lifecycle-events
https://reefjs.com/advanced/#batch-rendering

```
document.addEventListener('reef:render', function (event) {
	console.log('The UI was just updated inside this element.');
	console.log(event.target);
        if (event.target && event.target.closest('.myElement')) {
        
       }
});
```

Let me know if this doesn't answer your question. One method to achieve your desired outcome is to leverage event propagation to watch for a specific component being rendered.

**Comment by mouse0270** [comment]
> @mouse0270 (responding to the edit)
> 
> https://reefjs.com/api/#lifecycle-events https://reefjs.com/advanced/#batch-rendering
> 
> ```
> document.addEventListener('reef:render', function (event) {
> 	console.log('The UI was just updated inside this element.');
> 	console.log(event.target);
>         if (event.target && event.target.closest('.myElement')) {
>         
>        }
> });
> ```
> 
> Let me know if this doesn't answer your question. One method to achieve your desired outcome is to leverage event propagation to watch for a specific component being rendered.

the problem with `reef:render` is that it triggers when any component renders, I wanted an event similar to how store works 
```javascript
document.addEventListener('reef:render-Action', async (event) => { ... });
```

I basically solved this issue by doing
```javascript
// Refresh Select Pickers when Reactive Components are Rendered
document.addEventListener('reef:render', function (event) {
	// Create Control Specific Render Functions
	const id = event.target.id.replace('component', '');
	event.target.dispatchEvent(new CustomEvent(`reef:render-${id}`, {
		bubbles: true,
		detail: CONTROLS[id]
	}));
});
```
But this really only works because of how I setup my components.

**Comment by cferdinandi** [comment]
@mouse0270 Forgive me, as I haven't had a full cup of coffee yet today (☕️), but if I understand what you're saying correctly...

The `reef.store()` method fires multiple times when several pieces of data contained it are updated, _not_ that `reef.render()` renders the UI multiple times in response, correct?

**Comment by mouse0270** [comment]
Yes, so for example `document.addEventListener('reef:store-cart', function () { ... });` would fire on each update to the store.
```
cart.push('shirt');
cart.push('pants');
```
This would cause the conde inside of `reef:store-cart` event to be fired twice, once for each push. However... `document.addEventListener('reef:render', function (event) { ... });` only gets fired once. But the problem is that `reef:render` has no way to attach itself to a specific component. It gets called when any component renders.

I think `reef:render` is what I am looking for, as I want to run code once the component is done updating and not once the store is updated. But `reef:render` doesn't currently have component specific event listeners. Which in my example, you can see I sorta of got around, but it would be nice to have an official way to bind an event listener when a specific component renders.

**Comment by mouse0270** [comment]
Awe. I didn't realize the event was rendered on the component. I believe you mean as `reef:render-app` isn't valid.
```javascript
// Log whenever an element is rendered into
document.querySelector('#app').addEventListener('reef:render', function (event) {
	console.log('The UI was just for the #app component.');
	console.log(event.target);
});

// Log whenever an element is rendered into
document.querySelector('#app2').addEventListener('reef:render', function (event) {
	console.log('The UI was just for the #app2 component.');
	console.log(event.target);
});
```

That also works just fine, is it stated anywhere in the docs that the `reef:render` is triggered on the component? Did I just miss this? I really thought I read though the documents fairly well. but its a small thing I could be missing that.

**Comment by mouse0270** [comment]
Yup, just something small that I missed. Well call this closed. haha
